main.go

```diff
@@ -266,3 +289,94 @@ func getTaskContributorsHandler(w http.ResponseWriter, r *http.Request, db *sql.
 		return
 	}
 }
+
+// URL Funcs
+
+func urlFormHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
+		tmpl := template.Must(template.ParseFiles("url.html"))
+		tmpl.Execute(w, nil)
+	} else {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+	}
+}
+
+func submitURLHandler(db *sql.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodPost {
+			r.ParseForm()
+			urlInput := r.Form.Get("url")
```

Comment:
Please research the max length of a get request supported on browsers on linux/windows/macos and make sure you have good amount of space to store it.

I recommend hard coding a maximum size of the url and trim the rest.
if the len > SOME_CONST then warn the user that the url was trimmed.